app/code/Magento/Newsletter/Model/Subscriber.php

```diff
@@ -360,7 +360,6 @@ public function loadByCustomerId($customerId)
     {
         try {
             $customerData = $this->customerRepository->getById($customerId);
-            $customerData->setStoreId($this->_storeManager->getStore()->getId());
```

Comment:
We cannot simply remove the store_id check since it brings problems with multistore subscriptions. Details are here: https://github.com/magento/magento2/issues/10014